Add more detailed instructions to point release docs

getdave opened the pull request:
<!-- Learn the overall process and best practices for pull requests at https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/repository-management.md#pull-requests. -->

<!-- Gutenberg's license is in the process of updating to be dual-licensed under the GPL and MPL. As part of that transition, all new contributions are dual-licensed. For more information, see: https://github.com/WordPress/gutenberg/blob/trunk/LICENSE.md -->

## Description
I was recently trying to perform a point release of the Plugin but the docs (which were ironically written by me!) weren't clear enough.

This PR adds lots more detail and provides more specific step by step instructions that should (hopefully) be clear enough to follow without the requirement for "insider knowledge" from someone who has done a point release before.

## Testing Instructions

Review docs for accuracy. It's critical no steps are missed out or less than **100% clear**. We want contributors to feel confident they are not going to mess things up.

Also review spelling and grammar 🙏 

## Screenshots <!-- if applicable -->

## Types of changes
<!-- What types of changes does your code introduce?  -->
<!-- Bug fix (non-breaking change which fixes an issue) -->
<!-- New feature (non-breaking change which adds functionality) -->
<!-- Breaking change (fix or feature that would cause existing functionality to not work as expected) -->

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/ -->
- [ ] My code follows the accessibility standards. <!-- Guidelines: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/ -->
- [ ] I've tested my changes with keyboard and screen readers. <!-- Instructions: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/accessibility-testing.md -->
- [ ] My code has proper inline documentation. <!-- Guidelines: https://developer.wordpress.org/coding-standards/inline-documentation-standards/javascript/ -->
- [ ] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/HEAD/docs/contributors/code/native-mobile.md -->
- [ ] I've updated related schemas if appropriate. <!-- Reference: https://github.com/WordPress/gutenberg/tree/trunk/schemas -->

getdave committed:
Add more detailed instructions

getdave committed:
Add additional detail

ntsekouras commented on docs/contributors/code/release.md:
```diff
@@ -130,15 +130,33 @@ If you decide that the fixes deserve another release candidate before the stable
+As you proceed with the following process, it's worth bearing in mind that such point releases are not created as branches in their own right (e.g. [`release/12.5.0`](https://github.com/WordPress/gutenberg/tree/release/12.5)) but are simply [tags](https://github.com/WordPress/gutenberg/releases/tag/v12.5.1).
 
-**Please note** however, that if the previous release was an _RC_ (e.g. `x.y.0-rc.1`), the workflow will try to release the next major _stable_ version. Therefore, should you wish to release a point release for a stable version of the Plugin (e.g. `10.1.0`) but there is an _RC already published_ for the _next_ version of the Plugin (e.g. `10.2.0-rc.1`) you will need to manually select the stable  ...
+The method for point releases is nearly identical to the main Plugin release process (see above) but has some notable exceptions. Please be sure to read _the whole_ of this guide before proceeding.
```
Should we use `make` here? 🤔 

```suggestion
The method for point releases is nearly identical to the main Plugin release process (see above) but has some notable exceptions. Please make sure to read _the whole_ of this guide before proceeding.
```

ntsekouras commented on docs/contributors/code/release.md:
```diff
@@ -130,15 +130,33 @@ If you decide that the fixes deserve another release candidate before the stable
-Please note you **cannot create point releases for previous stable releases once a more recent stable release has been published** as this would require significant changes to how we upload plugin versions to the WP.org plugin SVN repo).
+The point release should only contain the _specific commits_ required. To do this you should checkout the previous _minor_ stable (i.e. non-RC) release (e.g. `release/12.5.0`) locally and then cherry pick any commits (from `trunk`) that you require to be included in the branch.
+
+If there are previous point releases (e.g. `12.5.1` .etc) these can be safely ignored as commits from those releases will have already been picked into the stable release branch as part of the previous point release process.
```
I think this might be more confusing as the process of `cherry picking` is explained above.. Maybe we should remove this paragraph (`If there are previous point releases....`)? 🤔

getdave replied:
I removed and simplified. Is it better?

ntsekouras commented on docs/contributors/code/release.md:
```diff
@@ -130,15 +130,33 @@ If you decide that the fixes deserve another release candidate before the stable
+Occasionally it's necessary to create a point release (i.e. X.Y._Z_) of the Plugin. This is most commonly to push fixes for regressions and/or bugs.
 
-The method for doing this is nearly identical to the main Plugin release process (see above). In the `Run workflow` control specify `stable` and, _if_ the previous version was stable (`x.y.z`), the workflow will release a point release, with z incremented (`x.y.(z+1)`) as necessary.
+As you proceed with the following process, it's worth bearing in mind that such point releases are not created as branches in their own right (e.g. [`release/12.5.0`](https://github.com/WordPress/gutenberg/tree/release/12.5)) but are simply [tags](https://github.com/WordPress/gutenberg/releases/tag/v12.5.1).
```
No need to point to a different thing here, as the `12.5.0` branch do not exists.
```suggestion
As you proceed with the following process, it's worth bearing in mind that such point releases are not created as branches in their own right (e.g. `release/12.5.0`) but are simply [tags](https://github.com/WordPress/gutenberg/releases/tag/v12.5.1).
```

ntsekouras reviewed (commented):
Thanks for the PR @getdave! 

This looks good to me! I've left a couple of small comments though.

getdave committed:
Use more appropriate wording

Co-authored-by: Nik Tsekouras <ntsekouras@outlook.com>

getdave committed:
Update docs/contributors/code/release.md

Co-authored-by: Nik Tsekouras <ntsekouras@outlook.com>

getdave committed:
Simplify and remove unwanted detail

ntsekouras commented on docs/contributors/code/release.md:
```diff
@@ -130,15 +130,31 @@ If you decide that the fixes deserve another release candidate before the stable
+#### Updating the release branch
 
-Please note you **cannot create point releases for previous stable releases once a more recent stable release has been published** as this would require significant changes to how we upload plugin versions to the WP.org plugin SVN repo).
+The point release should only contain the _specific commits_ required. To do this you should checkout the previous _minor_ stable (i.e. non-RC) release branch (e.g. `release/12.5.0`) locally and then cherry pick any commits that you require into that branch.
```
```suggestion
The point release should only contain the _specific commits_ required. To do this you should checkout the previous _minor_ stable (i.e. non-RC) release branch (e.g. `release/12.5`) locally and then cherry pick any commits that you require into that branch.
```

ntsekouras reviewed (approved):
Thanks Dave! Just one minor change and I think it's good to 🚢

getdave committed:
Use correct version in release branch.

Co-authored-by: Nik Tsekouras <ntsekouras@outlook.com>

getdave commented:
I'll leave this here today for other folks to add thoughts (if any) otherwise I'll look to merge tomorrow morning UTC.

chad1008 commented on docs/contributors/code/release.md:
```diff
@@ -130,15 +130,31 @@ If you decide that the fixes deserve another release candidate before the stable
+
+_If_ the previous release version was **stable** (`X.Y.Z` - e.g. `12.5.0`, `12.5.1` .etc) leave the the `Use workflow from` field as `trunk` and then specify `stable` in the text input field. The workflow will automatically create a point release, with z incremented (`x.y.(z+1)`) as required.
+
+_If_ however, the previous release was an **RC** (e.g. `X.Y.0-rc.1`) you will need to _manually_ select the _stable version's release branch_ (e.g. `12.5.0`) when creating the release. Failure to do this will cause the workflow to release the next major _stable_ version which is not what you want.
```
Doc looks great to me, this is the one section I'm not totally clear on. It's very clear on what to do when the previous version was an RC, but I'm completely sure what to expect if I were to fail to select the stable version's release branch (e.g. `12.5.0`).

> Failure to do this will cause the workflow to release the next major _stable_ version which is not what you want.

Does this mean that failing to select `v12.5.0` would ultimately release `12.6.0`, rather than the `12.5.1` patch that the contributor was hoping for? If so, I'd recommend moving the emphasis in the above quote from the word _stable_ to _major_ :)

getdave replied:
Thanks. Ok I've added in an example to illustrate this clearly. Yes if there is an RC then the workflow will try and release the stable version for that RC. So if you're RC is `12.6.0-rc.1` then running the workflow without taking due care would result in the release of `12.6.0` (stable).

getdave committed:
Add example to bring absolute clarity

Addresses https://github.com/WordPress/gutenberg/pull/38565#discussion_r801045734
